src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCM.java

```diff
@@ -108,21 +113,33 @@ public BitbucketSCM(
         }
         this.extensions.add(new BitbucketPostBuildStatus(scmHelper.getServerConfiguration().getId()));
         if (isBlank(projectKey) || isBlank(repositorySlug)) {
-            BitbucketRepository repository = scmHelper.getRepository(projectName, repositoryName);
-            cloneUrl = isBlank(cloneUrl) ? repository.getCloneUrls()
-                    .stream()
-                    .filter(link -> "http".equals(link.getName()))
-                    .findFirst()
-                    .map(BitbucketNamedLink::getHref)
-                    .orElse("") : cloneUrl;
+            BitbucketRepository repository;
+            if (!isBlank(mirrorName)) {
+                try {
+                    EnrichedBitbucketMirroredRepository mirroredRepository =
+                            descriptor.createMirrorHandler(scmHelper).fetchRepository(
+                                    new MirrorFetchRequest(id, credentialsId, projectName, repositoryName, mirrorName));
```

Comment:
This should be `serverId` not `id`. I'm not sure how this works as it is now

Reply:
It is serverId. Not sure why my Jenkins was working with id. I have fixed it.